### PR TITLE
[WIP] Tests without builtins fixtures

### PR DESCRIPTION
**Pull request by pkch**
Based on #3111. I am using this PR to examine the impact of removing builtins fixtures (# of failing tests and performance impact).

### CODE REVIEW

**Comment by pkch** [comment]
This is a branch on which I'm trying to make tests pass without builtins fixtures. I disabled units tests for testsemanal, testtypegen and testtransform because they kept parsing too many modules (I tried to fix it, but haven't completely). The only other tests that fail are pytest.

In pytest, I enabled the option (just for this PR) to ignore differences of error messages if they are in the same line (including duplicate messages in the same line).

With that, the current status is 72 failed, 2391 passed for pytest.

Obviously, many of the failing tests will disappear after a couple of small fixes (e.g., in check-incremental).

File: test-data/unit/check-flags.test [hunk 1]
```diff
@@ -194,7 +194,7 @@ x = 0  # type: NoReturn  # E: Incompatible types in assignment (expression has t
 [builtins fixtures/dict.pyi]
 
 [case testNoReturnImportFromTyping]
-from typing import NoReturn
```
Comment (ilevkivskyi) [inline test-data/unit/check-flags.test]: This is not right. If it fails, then typeshed should be updated, this is why this test is called ``testNoReturnImportFromTyping``.

File: test-data/unit/check-modules.test [hunk 2]
```diff
@@ -670,45 +670,45 @@ def f(x: str) -> None: pass
 [case testTypeCheckNamedModule]
 # cmd: mypy -m m.a
 [file m/__init__.py]
-None + 1
```
Comment (ilevkivskyi) [inline test-data/unit/check-modules.test]: Why do you replace ``None`` with ``object()`` everywhere? This does not seem right.

File: test-data/unit/check-isinstance.test [hunk 3]
```diff
@@ -434,13 +434,13 @@ def foo() -> Union[int, str]: pass
 
 x = foo()
 x = 1
-x = x + 1
```
Comment (ilevkivskyi) [inline test-data/unit/check-isinstance.test]: What is the reason for this change? This looks strange.

File: test-data/unit/check-modules.test (hunk 2 above)
Comment (pkch) [inline test-data/unit/check-modules.test]: I only replaced the cases where the intent is to check whether mypy gets to a given line. The test basically assumes that `None + 1` will cause type error. But they don't fail because `--strict-optional` isn't the default. `object() + 1` will fail under any conditions, so I thought I'd use it.

File: test-data/unit/check-isinstance.test (hunk 3 above)
Comment (pkch) [inline test-data/unit/check-isinstance.test]: Because of #3122. This test doesn't actually care about that behavior, it simply wanted to see if `x + 1` is allowed. But until #3122 is fixed, the side effect is that `x` changes type in a inconsistent manner, making the test incorrect without fixtures.

File: test-data/unit/check-flags.test (hunk 1 above)
Comment (pkch) [inline test-data/unit/check-flags.test]: But python runtime would crash on that. Why would we want to allow it in mypy?

Comment (ilevkivskyi) [inline test-data/unit/check-flags.test]: ``NoReturn`` is in ``typing`` now.

Comment (pkch) [inline test-data/unit/check-flags.test]: Oh I'm on python 3.6.0 so didn't realize that haha

File: test-data/unit/check-modules.test (hunk 2 above)
Comment (ilevkivskyi) [inline test-data/unit/check-modules.test]: IIRC, some tests are actually run with automatic addition of ``--strict-optional`` flag. Why this changes with your PR?

Comment (pkch) [inline test-data/unit/check-modules.test]: It shouldn't change. 

But apart from flags added to tests, and tests *inside* typeshed, and `mypy mypy` tests, I can't find tests that have automatic `strict-optional`. Are you sure it was done to the pytest tests?

**Comment by pkch** [comment]
The main purpose of this PR is to allow everyone to run no-fixtures tests locally by setting `NO_BUILTINS_FIXTURES` env. var. I removed the Travis setting that made this happen automatically.

**Comment by pkch** [comment]
After you review, I was going to rebase this off the current master to make it less likely to cause weird merge conflicts.

File: test-data/unit/check-modules.test (hunk 2 above)
Comment (ilevkivskyi) [inline test-data/unit/check-modules.test]: > Are you sure it was done to the pytest tests?

I am not, but I assumed this, since the tests were passing before, i.e. were giving an error for ``None + 1``.

Comment (pkch) [inline test-data/unit/check-modules.test]: It was just because in the builtins fixture `int.__radd__` isn't defined (only `int.__add__`). And the subtyping is only checked for the rhs, not for the lhs of these operators.

File: test-data/unit/check-flags.test (hunk 1 above)
Comment (pkch) [inline test-data/unit/check-flags.test]: BTW, you mean `typing.NoReturn` is in cpython master right? It's not yet in 3.6.1. I think it's fine for mypy to allow things that are valid on cpython master.

Comment (ilevkivskyi) [inline test-data/unit/check-flags.test]: It will be included in 3.6.2

**Comment by pkch** [comment]
This PR has a lot of old stuff that would make merging it very messy. I'm closing this in favor of #3268. @ilevkivskyi  I'm not just rebasing on master within this PR because you were reviewing it, and you'll lose the context if I do so.
